src/Microsoft.HttpRepl.IntegrationTests/Mocks/MockedFileSystem.cs

```diff
@@ -0,0 +1,127 @@
+        {
+            // Setup the "files" that should exist in the file system for the various tests here.
+            { $"{nameof(PostCommandTests)}-{nameof(PostCommandTests.ExecuteAsync_MultiPartRouteWithBodyFromFile_VerifyResponse)}.txt", new MockedFile("Test Post Body From File") },
+            { $"{nameof(PutCommandTests)}-{nameof(PutCommandTests.ExecuteAsync_MultiPartRouteWithBodyFromFile_VerifyResponse)}.txt", new MockedFile("Test Put Body From File") },
+            { $"{nameof(PatchCommandTests)}-{nameof(PatchCommandTests.ExecuteAsync_MultiPartRouteWithBodyFromFile_VerifyResponse)}.txt", new MockedFile("Test Patch Body From File") }
+        };
+
+        internal string GetFile(string path)
+        {
+            return _files[path].Content;
+        }
+
+        internal void SetFile(string path, string contents)
+        {
+            _files[path].Content = contents;
+        }
+
+        public Stream Create(string path)
+        {
+            _files[path] = new MockedFile("");
+            return _files[path].Stream;
+        }
+
+        public void Delete(string path)
+        {
+            if (_files.ContainsKey(path))
+            {
+                _files.Remove(path);
+            }
+        }
+
+        public bool Exists(string path)
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                return false;
+            }
+
+            return _files.ContainsKey(path);
+        }
+
+        public byte[] ReadAllBytes(string path)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException(path);
+            }
+
+            if (!Exists(path))
+            {
+                throw new FileNotFoundException();
+            }
+
+            return _files[path].Stream.ToArray();
+        }
+
+        public string[] ReadAllLines(string path)
+        {
+            string alltext = _files[path].Content;
+            return alltext.Split(new string[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
+        }
+
+        public void WriteAllLines(string path, IEnumerable<string> contents)
+        {
+            _files[path] = new MockedFile(string.Join(Environment.NewLine, contents));
+        }
+
+        public void WriteAllText(string path, string contents)
+        {
+            _files[path].Content = contents;
+        }
+
+        public string GetTempFileName()
+        {
+            string path = GetRandomFileName();
+            _files[path].Content = "";
+            return path;
+        }
+
+        private string GetRandomFileName()
+        {
+            RNGCryptoServiceProvider rng = new RNGCryptoServiceProvider();
+            byte[] bytes = new byte[9];
+            rng.GetBytes(bytes);
+            string path = Convert.ToBase64String(bytes);
+            return path;
+        }
+    }
+
+    internal class MockedFile
+    {
+        public MemoryStream Stream { get; private set; }
```

Comment:
You're gonna have some isuses with a Stream as your backing store.  For example, this fails:

    var fs = new MockedFileSystem();
    using (Stream s = fs.Create(@"C:\doesnotexist\file.txt"))
    using (StreamWriter sw = new StreamWriter(s))
    {
        sw.WriteLine("test");
    }

because the stream is not expandable.  You'd also run into issues if the stream was closed and then tried to be read from again later.

Reply:
One approach: don't expose Stream from the IFileSystem - do everything with content (`string` or `byte[]`).

If you need to expose Stream, then the mock should probably use a `byte[]` buffer as a backing store, and open a stream to that - it'll give you a block of space to write to for tests, just don't go over the buffer size.

Reply:
If I can get rid of the usage of Stream, that'd definitely make things simpler - that one function (Create) added a lot of complexity to that implementation. I'll take a look and see what that'd look like.

Reply:
Alright, I was able to remove it, though I'm not super thrilled with the replacement. The original code has an awkward dance around whether the headers and body are written to the same file, different files, no files, or only one of them is written to a file... and it wrote it out to a stream regardless, even when not writing it to a file. I'll resolve this for now, since the MemoryStream-backed MockedFile is gone, but I'm guessing there'll be comments on the new code.